core/src/main/java/org/opensearch/sql/calcite/CalciteRexNodeVisitor.java

```diff
@@ -116,6 +117,32 @@ public RexNode visitLiteral(Literal node, CalcitePlanContext context) {
     }
   }
 
+  public static SpanUnit intervalUnitToSpanUnit(IntervalUnit unit) {
+    return switch (unit) {
+      case MICROSECOND -> SpanUnit.MILLISECOND;
+      case SECOND -> SpanUnit.SECOND;
+      case MINUTE -> SpanUnit.MINUTE;
+      case HOUR -> SpanUnit.HOUR;
+      case DAY -> SpanUnit.DAY;
+      case WEEK -> SpanUnit.WEEK;
+      case MONTH -> SpanUnit.MONTH;
+      case QUARTER -> SpanUnit.QUARTER;
+      case YEAR -> SpanUnit.YEAR;
+      case UNKNOWN -> SpanUnit.UNKNOWN;
+      default -> throw new UnsupportedOperationException("Unsupported interval unit: " + unit);
+    };
+  }
+
+  @Override
+  public RexNode visitInterval(
```

Comment:
duplicated to https://github.com/opensearch-project/sql/pull/3446/files#diff-e5198d773af75bf3173ef25676a2803a0091cb51e32d6ae30241273519d30261R123. Can remove this when #3446 merged.